backend/orchestrators/Orchestrator.py

```diff
@@ -404,6 +418,7 @@ def stream_with_data(self, body, headers, endpoint, message_uuid, history_metada
 
                         if role == "tool":
                             response["choices"][0]["messages"].append(lineJson["choices"][0]["messages"][0]["delta"])
+                            self.conversation_client.update_conversation_item(response, history_metadata["conversation_id"])
```

Comment:
It looks like all the calls that leverage `update_conversation_item` are in the same method (`stream_with_data`). What about the other methods for responding like `stream_without_data`?